vcd/resource_vcd_vapp.go

```diff
@@ -396,14 +396,26 @@ func resourceVcdVAppRead(d *schema.ResourceData, meta interface{}) error {
 		return nil
 	}
 
-	if _, ok := d.GetOk("template_name"); ok {
-		ip, err := getVAppIPAddress(d, meta)
-		if err != nil {
-			return err
+	if _, ok := d.GetOk("ip"); ok {
+		ip := "allocated"
+
+		oldIp, newIp := d.GetChange("ip")
+
+		log.Printf("[DEBUG] IP has changes, old: %s - new: %s", oldIp, newIp)
+
+		if newIp != "allocated" {
+			log.Printf("[DEBUG] IP is assigned. Lets get it (%s)", d.Get("ip"))
+			ip, err = getVAppIPAddress(d, meta)
+			if err != nil {
+				return err
+			}
+		} else {
+			log.Printf("[DEBUG] IP is 'allocated'")
 		}
+
 		d.Set("ip", ip)
```

Comment:
I'd argue it might make more sense for Debug purposes to move [this Print statement](https://github.com/terraform-providers/terraform-provider-vcd/pull/37/files#diff-337511eab4aa431ec392f2d7cf553b19R413) here?